### PR TITLE
Adds a Dockerfile describing a CORS proxy for the platform app

### PR DESCRIPTION
**Pull request by kenoir**
This is in order to provide CORS config outside of the Finatra
application, and to account for necessary ALB URL rewrites.

Ultimately this allows in-browser clients to make requests of the API
directly.

### CODE REVIEW

File: docker/nginx/api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM nginx:alpine
+COPY nginx.conf /etc/nginx/nginx.conf
```
Comment (alexwlchan) [inline docker/nginx/api/Dockerfile]: Do we want to add any metadata to the images? e.g. maintainer, version, description

https://docs.docker.com/engine/reference/builder/#label

File: docker/nginx/api/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+               add_header 'Access-Control-Allow-Origin' '*';
+               add_header 'Access-Control-Allow-Credentials' 'true';
+               add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+               add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+            }
+            if ($request_method = 'GET') {
+               add_header 'Access-Control-Allow-Origin' '*';
+               add_header 'Access-Control-Allow-Credentials' 'true';
+               add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+               add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
```
Comment (alexwlchan) [inline docker/nginx/api/nginx.conf]: Could we pull the four lines above out of the `if` statements, and just have a few custom lines in an

```nginx
if ($request_method = 'OPTIONS')
```

block?

File: terraform/services/ecs_tasks/variables.tf [hunk 3]
```diff
@@ -7,8 +7,13 @@ variable "template_name" {
   default     = "default"
 }
 
-variable "image_uri" {
-  description = "URI of container image for primary container"
+variable "nginx_uri" {
+  description = "URI of container image for nginx"
+  default     = "library/hello-world"
```
Comment (alexwlchan) [inline terraform/services/ecs_tasks/variables.tf]: Should this have a default? Do we ever not want it to be explicitly specified?

File: terraform/services/ecs_tasks/variables.tf [hunk 4]
```diff
@@ -7,8 +7,13 @@ variable "template_name" {
...
+}
+
+variable "app_uri" {
+  description = "URI of container image for app"
   default     = "library/hello-world"
```
Comment (alexwlchan) [inline terraform/services/ecs_tasks/variables.tf]: Ditto above.

File: docker/nginx/api/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+LABEL description="nginx with config for api (adds CORS headers)"
```
Comment (alexwlchan) [inline docker/nginx/api/Dockerfile]: 👍

File: docker/nginx/api/nginx.conf [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      add_header 'Access-Control-Allow-Credentials' 'true';
+      add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+      add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+
+      if ($request_method = 'OPTIONS') {
+        add_header 'Access-Control-Max-Age' 1728000;
+        add_header 'Content-Type' 'text/plain charset=UTF-8';
+        add_header 'Content-Length' 0;
+        return 204;
+      }
```
Comment (alexwlchan) [inline docker/nginx/api/nginx.conf]: 👍

File: docker/nginx/services/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+http {
+    server {
+        listen 9000;
+        location ~ ^/[^/]+/.* {
+            rewrite ^/[^/]+/(.*) /$1 break;
+            proxy_pass http://app:8888;
+        }
+    }
+}
```
Comment (alexwlchan) [inline docker/nginx/services/nginx.conf]: nit: 4 space indents here, 2 space indents in the other config file?

File: docker/nginx/api/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+FROM nginx:alpine
```
Comment (jamesgorrie) [inline docker/nginx/api/Dockerfile]: Why `alpine`? Just because it's more lightweight / efficient?

Comment (kenoir) [inline docker/nginx/api/Dockerfile]: yes.

**Pull request merged**